homeassistant/components/nina/__init__.py

```diff
@@ -147,6 +162,21 @@ def _parse_data(self) -> dict[str, list[NinaWarningData]]:
                 if re.search(
                     self.headline_filter, raw_warn.headline, flags=re.IGNORECASE
                 ):
+                    _LOGGER.debug(
+                        f"Ignore warning ({raw_warn.id}) by headline filter ({self.headline_filter}) with headline: {raw_warn.headline}"
+                    )
+                    continue
+
+                affected_areas_string: str = ", ".join(
+                    [str(area) for area in raw_warn.affected_areas]
+                )
+
+                if not re.search(
```

Comment:
So this doesn't actually filter the result more than making a `True` or `False`?

Should it not be saved to coordinator results so the `binary_sensor` can trim down the result further?

Reply:
Yes, this is the intended function. If the filter detects a region the message is forwarded, if not it is ignored.
I don't see why the message should be filtered further in binary_sensor.

Reply:
If you're happy with it, please click "Ready for review".
I see a doc PR has been made